examples/helpers/widgets.rs

```diff
@@ -76,93 +76,90 @@ pub fn spawn_option_button<T>(
+                BUTTON_BORDER_RADIUS_SIZE
+            } else {
+                px(0)
+            })
+            .with_right(if is_last {
+                BUTTON_BORDER_RADIUS_SIZE
+            } else {
+                px(0)
+            }),
+        BackgroundColor(bg_color),
+        RadioButton,
+        WidgetClickSender(option_value.clone()),
+        children![(
+            ui_text(option_name, fg_color),
+            RadioButtonText,
+            WidgetClickSender(option_value),
+        )],
+    )
 }
 
 /// Spawns the buttons that allow configuration of a setting.
 ///
 /// The user may change the setting to any one of the labeled `options`. The
 /// value of the given type parameter will be packaged up and sent as a
 /// [`WidgetClickEvent`] when one of the radio buttons is clicked.
-pub fn spawn_option_buttons<T>(
-    parent: &mut ChildSpawnerCommands,
-    title: &str,
-    options: &[(T, &str)],
-) where
+pub fn option_buttons<T>(title: &str, options: &[(T, &str)]) -> impl Bundle
+where
     T: Clone + Send + Sync + 'static,
 {
+    let buttons = options
+        .iter()
+        .cloned()
+        .enumerate()
+        .map(|(option_index, (option_value, option_name))| {
+            option_button(
+                option_value,
+                option_name,
+                option_index == 0,
+                option_index == 0,
+                option_index == options.len() - 1,
+            )
+        })
+        .collect::<Vec<_>>();
     // Add the parent node for the row.
-    parent
-        .spawn(Node {
+    (
+        Node {
             align_items: AlignItems::Center,
             ..default()
-        })
-        .with_children(|parent| {
-            spawn_ui_text(parent, title, Color::BLACK).insert(Node {
-                width: px(125),
-                ..default()
-            });
-
-            for (option_index, (option_value, option_name)) in options.iter().cloned().enumerate() {
-                spawn_option_button(
-                    parent,
-                    option_value,
-                    option_name,
-                    option_index == 0,
-                    option_index == 0,
-                    option_index == options.len() - 1,
-                );
-            }
-        });
+        },
+        Children::spawn((
+            Spawn((
+                ui_text(title, Color::BLACK),
+                Node {
+                    width: px(125),
+                    ..default()
+                },
+            )),
+            SpawnIter(buttons.into_iter()),
+        )),
+    )
 }
 
-/// Spawns text for the UI.
-///
-/// Returns the `EntityCommands`, which allow further customization of the text
-/// style.
-pub fn spawn_ui_text<'a>(
-    parent: &'a mut ChildSpawnerCommands,
-    label: &str,
-    color: Color,
-) -> EntityCommands<'a> {
-    parent.spawn((
+/// Creates a text bundle for the UI.
+pub fn ui_text(label: &str, color: Color) -> (Text, TextFont, TextColor) {
```

Comment:
```suggestion
pub fn ui_text(label: &str, color: Color) -> impl Bundle {
```
Would it be better to declare intention here? There's no need for it to have the concrete type.

Reply:
I think I did this because I was fighting lifetimes? Now I'm not sure anymore..